### DISCUSSION TITLE
QoS / Kafka Topic Prioritization w/ ClickHouse Kafka Connect Sink

### DISCUSSION THREAD

**Discussion by hodgesz** [ClickHouse/ClickHouse#60589] [Q&A]
Our use case is using Kafka and an intermediate step for our events before persisting in ClickHouse to offer backpressure and avoid causing performance issues on the ClickHouse cluster for other transactions. We are currently using self-managed ClickHouse and Kafka clusters.

We have no issues publishing our events to Kafka, but curious if we are able to use the CH sink (https://www.confluent.io/hub/clickhouse/clickhouse-kafka-connect) as a way to move these over to CH. We have one tricky requirement where certain events/topics need to take priority over others, sort of like quality of service.

Do you know how we could approach using the sink to satisfy this requirement? We were thinking of maybe running multiple instances of the sink with different configurations to try to satisfy the different priorities or QoS across the events/topics.

As a fallback solution, we could certainly write our own custom Kafka consumer, but we are trying to avoid this complexity if possible. Any thoughts or ideas?

**Comment by Paultagoras** [comment]
Closing this here as the discussion is taking place at https://github.com/ClickHouse/clickhouse-kafka-connect/discussions/337
